error[E0599]: no method named `and_utc` found for struct `NaiveDateTime` in the current scope

### Bug Description

Compiler error

### Steps to Reproduce

1. Upgrade pyo3 from 0.19.2 to 0.20.0
2. Compile

### Backtrace

```shell
error[E0599]: no method named `and_utc` found for struct `NaiveDateTime` in the current scope
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pyo3-0.20.0/src/conversions/chrono.rs:230:15
    |
230 |         Ok(dt.and_utc())
    |               ^^^^^^^ method not found in `NaiveDateTime`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `pyo3` (lib) due to previous error
```
```


### Your operating system and version

Linux

### Your Python version (`python --version`)

N/A

### Your Rust version (`rustc --version`)

Rust 1.73

### Your PyO3 version

0.20.0

### How did you install python? Did you use a virtualenv?

N/A

### Additional Info

https://github.com/brianmay/robotica-rust/actions/runs/6506697314/job/17672737275?pr=277

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -33,7 +33,7 @@ inventory = { version = "0.3.0", optional = true }
 
 # crate integrations that can be added using the eponymous features
 anyhow = { version = "1.0", optional = true }
-chrono = { version = "0.4", default-features = false, optional = true }
+chrono = { version = "0.4.25", default-features = false, optional = true }
 eyre = { version = ">= 0.4, < 0.7", optional = true }
 hashbrown = { version = ">= 0.9, < 0.15", optional = true }
 indexmap = { version = ">= 1.6, < 3", optional = true }

--- a/newsfragments/3512.fixed.md
+++ b/newsfragments/3512.fixed.md
@@ -1,0 +1,1 @@
+Fix minimum version specification for optional `chrono` dependency

